lib/src/bluez_client.dart

```diff
@@ -657,6 +664,247 @@ class BlueZAdvertisement extends DBusObject {
   }
 }
 
+/// Management for Battery provisioning
+class BlueZBatteryProviderManager {
+  final String _batteryProviderManagerInterfaceName =
+      'org.bluez.BatteryProviderManager1';
+
+  final BlueZClient _client;
+  final _BlueZObject _object;
+  int _nextBatteryProviderId;
+  int _nextBatteryId;
+
+  BlueZBatteryProviderManager(this._client, this._object)
+      : _nextBatteryProviderId = 0,
+        _nextBatteryId = 0;
+
+  /// Registers a new battery with an optional label.
+  Future<BlueZBattery> registerBattery(String? name) async {
+    var device = BlueZBattery(
+        DBusObjectPath('/org/bluez/battery/device_$_nextBatteryId'),
+        source: name);
+    _nextBatteryId += 1;
+
+    await _client._bus.registerObject(device);
+    return device;
+  }
+
+  /// Unregisters the battery
+  Future<void> unregisterBattery(BlueZBattery device) async {
+    await _client._bus.unregisterObject(device);
+  }
+
+  /// Registers a battery provider. A registered battery provider can
+  /// then expose objects with org.bluez.BatteryProvider(5) interface.
```

Comment:
The documentation talks about the DBus interfaces, which doesn't seem appropriate to this library (i.e. this is an internal detail). Reading https://github.com/hadess/bluez/blob/master/doc/battery-api.txt it's really unclear to me when you use registerBattery and when you use registerBatteryProvider - do you understand the difference? Can the documentation be updated to reflect how these functions are to be used?

Reply:
`registerBattery` is used before `registerBatteryProvider` so you can provide the battery you register.

Reply:
Clarified in doc comments.